src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -340,10 +325,10 @@ default MemorySegment allocateFrom(ValueLayout.OfDouble elementLayout, double...
         return copyArrayWithSwapIfNeeded(elements, elementLayout);
     }
 
+    @ForceInline
     private MemorySegment copyArrayWithSwapIfNeeded(Object array, ValueLayout elementLayout) {
         int size = Array.getLength(Objects.requireNonNull(array));
-        Objects.requireNonNull(elementLayout);
-        MemorySegment segment = allocateNoInit(elementLayout.byteSize() * size, elementLayout.byteAlignment());
+        MemorySegment segment = allocateNoInit(Objects.requireNonNull(elementLayout), size);
         if (size > 0) {
```

Comment:
Do we need this check? Is it not so that a vast majority of provided arrays would have a size > 0 and in the rare occasion there is a zero-length array, it will still work without this check?

Reply:
The check was needed to preserve semantics of the method (there was one test failure because of that - but I think we can fix the test instead).